packages/gatsby-theme-project-portal/gatsby-node.js

```diff
@@ -129,3 +193,78 @@ exports.createPages = async ({ graphql, actions, reporter }) => {
     })
   })
 }
+
+exports.onPreBuild = async ({ reporter, basePath, pathPrefix, graphql }) => {
+  const result = await graphql(`
+    query {
+      allProject {
+        nodes {
+          title
+          agency
+          topics {
+            title
+          }
+          slug
+          summary
+          statusOfData
+          status
+          startDate
+          requirement
+          question
+          purpose
+          projectTeam {
+            name
+            employer
+          }
+          priorResearch
+          opportunityCloses
+          mainContact {
+            name
+          }
+          fundingInfo
+          expertise
+          faq {
+            text
+            title
+          }
+          deliverable
+          emailContent
+          endDate
+          slug
+        }
+      }
+      allGeneralPage {
+        nodes {
+          slug
+          lede
+          faq {
+            text
+            title
+          }
+          aims {
+            text
+            title
+          }
+          title
+        }
+      }
+    }
+  `)
```

Comment:
Could this be moved to a graphQL fragment in the utils/search.js file? That's where it's actually "used".
This might not be possible right now – the utils isn't in the `src` directory (though why it isn't is a bit beyond me, no idea why I did that) so Gatsby might not recognize fragments there.

Reply:
A fragment doesn't work, but what about adding the query in the module's exports?

```
const searchQuery = `
    query {
      allProject {
        nodes {
          title
          agency
          topics {
            title
          }
          slug
          summary
          statusOfData
          status
          startDate
          requirement
          question
          purpose
          projectTeam {
            name
            employer
          }
          priorResearch
          opportunityCloses
          mainContact {
            name
          }
          fundingInfo
          expertise
          faq {
            text
            title
          }
          deliverable
          emailContent
          endDate
          slug
        }
      }
      allGeneralPage {
        nodes {
          slug
          lede
          faq {
            text
            title
          }
          aims {
            text
            title
          }
          title
        }
      }
    }
  `

module.exports = { createSearchIndex, searchQuery }
```